var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -490,12 +499,20 @@ def bjam_python_line(self, spec):
         if spec.satisfies("@:1.58"):
             return ""
 
-        return "using python : {0} : {1} : {2} : {3} ;\n".format(
-            spec["python"].version.up_to(2),
-            spec["python"].command.path,
-            spec["python"].headers.directories[0],
-            spec["python"].libs[0],
-        )
+        if spec.satisfies("platform=windows"):
+            return "using python : {0} : {1} : {2} : {3} ;\n".format(
+                spec["python"].version.up_to(2),
+                spec["python"].command.path.as_posix(),
+                spec["python"].prefix.include.as_posix(),
+                spec["python"].prefix.libs.as_posix(),
+            )
+        else:
+            return "using python : {0} : {1} : {2} : {3} ;\n".format(
+                spec["python"].version.up_to(2),
+                spec["python"].command.path,
+                spec["python"].headers.directories[0],
+                spec["python"].libs[0],
```

Comment:
The first two entries of each of these are the same (or in this second case I think you can use `spec["python"].command.path.as_posix()`).

The third item here would tend to be more robust as it's expressed in the non-windows case: does it return something that isn't usable for you on Windows (after converting to posix)?

The fourth item does seem significantly different.

Overall would it be possible to consolidate like:

```
if windows:
  include = 
else
  include = 
# Likewise for the fourth item
...
return "using python...".format(
  version,
  command.path.as_posix()
  include,
  ...
)
```

Reply:
I was able to put that method back to essentially the original by fixing how the python package file looks for the libs directory on windows instead.